test/framework/src/main/java/org/elasticsearch/datageneration/queries/QueryGenerator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.datageneration.queries;
+
+import org.apache.lucene.search.join.ScoreMode;
+import org.elasticsearch.core.Booleans;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class QueryGenerator {
+
+    private final Map<String, Map<String, Object>> mappingLookup;
+    private final Map<String, Object> mappingRaw;
+
+    public QueryGenerator(Map<String, Map<String, Object>> mappingLookup, Map<String, Object> mappingRaw) {
+        this.mappingLookup = mappingLookup;
+        this.mappingRaw = mappingRaw;
+    }
+
+    public List<QueryBuilder> generateQueries(String path, Map<String, Object> mapping, Object value) {
+        // This query generator cannot handle fields with periods in the name.
+        if (path.equals("host.name")) {
+            return List.of();
+        }
+        if (mapping == null || isEnabled(path) == false) {
+            return List.of();
+        }
+        boolean isIndexed = (Boolean) mapping.getOrDefault("index", true);
```

Comment:
It feels like this is for the leaf to decide? I am pretty sure you can do some queries using doc_values, they are just slower.